core/spec/models/spree/product_spec.rb

```diff
@@ -11,6 +11,11 @@ class Extension < Spree::Base
 
 describe Spree::Product, :type => :model do
 
+  context 'associations' do
+    it { should have_many(:promotions).class_name('Spree::Promotion').through(:promotion_rules) }
```

Comment:
@Bharat311 - Please use is_expected instead of should, everything else looks good to me.